### ISSUE TITLE
Will LSTMBlockFusedCell be supported in tensorflow 2.0?

### ISSUE THREAD

**Issue by npuichigo** [tensorflow/tensorflow#26642]
I noticed that keras in tensorflow 2.0 unifies backend between CPU and GPU with CuDNN, but it's a little slow on CPU. Will you add LSTMBlockFusedCell or LSTMBlockCell to tf 2.0, since they have better performance?

**Comment by qlzh727** [comment]
Thanks for checking. I think LSTMBlockFusedCell or LSTMBlockCell won't be in the core API for 2.0. Having said that, I could update the cpu kernel to be a more performant version like LSTMBlockCell. The only down side I can see is that user will loss the python level debugability since it will be one fused op. 

I will discuss the trade off within teams and post it here.

**Comment by qlzh727** [comment]
Adding @protoget who previously worked on cudnn kernels.

**Comment by bhushanbrb** [comment]
Many of the good functions from **tf.contrib** are eliminated, such as CRF conditional random field etc how can one use these packages ( by using source code from these packages)? @qlzh727 
any suggestions pls

**Comment by andportnoy** [comment]
@qlzh727 Any updates on this?

**Comment by maziyarpanahi** [comment]
How can we find the author(s) and see if they are willing to support these two in Addons?

**Comment by qlzh727** [comment]
Sorry for the very late reply. There has been some work to add the fused c kernel to TF, and they haven't be expose to python side yet. I have some other issue to work with at the moment, will take a closer look within this month.

**Comment by maziyarpanahi** [comment]
Thanks @qlzh727 we appreciate your work. 
I would be fine if `LSTMBlockFusedCell` and `TimeReversedFusedRNN` move to `addons` or even willing to do the work and add them myself if I can find a simple instruction as to how to move something from `contrib` to `addons`.

**Comment by andportnoy** [comment]
@qlzh727 Thank you for your reply!

**Comment by npuichigo** [comment]
@qlzh727 Since we can benefit from the python level debugability when using keras.LSTMCell and keras.RNN, I think improve the performance of LSTM/GRU with a single fused op is possible.

In PyTorch, CuDNN is used for LSTM on GPU, and it will fall back to a fused lstm kernel when using CPU. So is it possible to integrate the LSTMBlockFusedCell into keras.LSTM when CPU is available.

(https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/layers/recurrent_v2.py#L910) vs (https://pytorch.org/docs/stable/_modules/torch/nn/modules/rnn.html#LSTM)

**Comment by bparr** [comment]
I believe the underlying LSTMBlockFusedCell op is already available in TF 2.0 as https://www.tensorflow.org/api_docs/python/tf/raw_ops/BlockLSTM. See https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/contrib/rnn/python/ops/lstm_ops.py#L654 for more information on how it is used. The only difference I see is the default cell_clip value is now 3.0. Also, there is a BlockLSTMV2, which is different somehow (https://www.tensorflow.org/api_docs/python/tf/raw_ops/BlockLSTMV2).

**Comment by rmothukuru** [comment]
@npuichigo,
Can you please respond to the above comment? Thanks!

**Comment by npuichigo** [comment]
@rmothukuru Sorry for the late reply. I checked the recent status and it seems LSTMBlockFusedCell is only exposed in raw_ops. LSTMCell in keras still doesn't use a fused kernel, but I don't know whether the performance of LSTMCell with tf.function is on par with a fused kernel.

**Comment by mohantym** [comment]
Hi @npuichigo!
We are checking to see whether you still need help in this issue .Have you checked this  [document on LSTMCell](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/rnn_cell/LSTMCell) from TF 2.7 yet?

**Comment by npuichigo** [comment]
have switched to pytorch for a long time. no need for help currently

**Comment by coleary-hyperscience** [comment]
> Hi @npuichigo! We are checking to see whether you still need help in this issue .Have you checked this [document on LSTMCell](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/rnn_cell/LSTMCell) from TF 2.7 yet?

I hope I am not missing anything obvious, but doesn't the recommendation in that link (i.e. `Please use ... tf.contrib.rnn.LSTMBlockFusedCell for better performance on CPU`) require you to not be using tensorflow 2.0+? Why is this the recommendation on the (currently 2.8) version of the documentation when it isn't supported?

**Comment by npuichigo** [comment]
@coleary-hyperscience I always want a LSTM implementation which is a fused kernel on CPU and can seamlessly switch to CuDNN on GPU. The recommendations here don't meet my needs, but PyTorch LSTM does.

As for the document, I think it's out of date since there's no more tf.contrib anymore. It was the original motive of this issue, to find an alternative in tensorflow 2.0

**Comment by mohantym** [comment]
Ok @npuichigo ! Reopening as it still seems to be a valid feature request . Thank you!

**Comment by gowthamkpr** [comment]
@npuichigo Discussed it with the team and this cannot be done as this is a niche case. Thanks!
